config: tests with vitest fail when running on Windows

Verdaccio: latest
node: 20.15.1
npm: 10.8.2

Used to work fine with jest but now test in `config` package fail when running on Windows:

```bash
cd packages/config
pnpm test

> @verdaccio/config@7.0.0-next-7.20 test C:\Github\~verdaccio\verdaccio\packages\config
> vitest run

 RUN  v2.0.4 C:/Github/~verdaccio/verdaccio/packages/config

 · test/config.spec.ts (19)
 ✓ test/package-access.spec.ts (11)
 ❯ test/config.path.spec.ts (6)
   ❯ config-path (6)
 ✓ test/agent.spec.ts (4)
 ✓ test/builder.spec.ts (3)
 ✓ test/config-parsing.spec.ts (7)
 ✓ test/config-utils.spec.ts (4)
 ❯ test/config.path.spec.ts (6)
   ❯ config-path (6)
     ❯ findConfigFile (6)
       ❯ using defiled location from arguments (1)
         × with custom location
       ❯ whith env variables (4)
         × the env XDG_CONFIG_HOME is defined and the directory exist but config file is missing
         × with HOME if directory exist but config file is missing
         ❯ error handling (2)
           × XDG_CONFIG_HOME is not directory fallback to default
           ✓ no permissions on read default config file
       ❯ with no env variables (1)
         × with relative location
 ✓ test/config.spec.ts (19)
 ✓ test/package-access.spec.ts (11)
 ✓ test/token.spec.ts (1)
 ✓ test/uplinks.spec.ts (7)
 ✓ test/utils.spec.ts (2)

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯ Failed Tests 5 ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯
 FAIL  test/config.path.spec.ts > config-path > findConfigFile > using defiled location from arguments > with custom location 
AssertionError: expected 'C:\home\user\custom\location\config.y…' to deeply equal '/home/user/custom/location/config.yaml'    

- Expected
+ Received

- /home/user/custom/location/config.yaml
+ C:\home\user\custom\location\config.yaml

 ❯ test/config.path.spec.ts:40:74
     38|         mkdirSyncMock.mockReturnValue(true);
     39|         writeFileSyncMock.mockReturnValue(undefined);
     40|         expect(findConfigFile('/home/user/custom/location/config.yaml')).toEqual(
       |                                                                          ^
     41|           '/home/user/custom/location/config.yaml'
     42|         );

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/5]⎯
 FAIL  test/config.path.spec.ts > config-path > findConfigFile > whith env variables > the env XDG_CONFIG_HOME is defined and 
the directory exist but config file is missing
AssertionError: expected '\home\user\verdaccio\config.yaml' to deeply equal '/home/user/verdaccio/config.yaml'

- Expected
+ Received

- /home/user/verdaccio/config.yaml
+ \home\user\verdaccio\config.yaml

 ❯ test/config.path.spec.ts:57:34
     55|         vi.stubEnv('XDG_CONFIG_HOME', '/home/user');
     56| 
     57|         expect(findConfigFile()).toEqual('/home/user/verdaccio/config.yaml');
       |                                  ^
     58|         expect(writeFileSyncMock).toHaveBeenCalledWith(
     59|           '/home/user/verdaccio/config.yaml',

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/5]⎯
 FAIL  test/config.path.spec.ts > config-path > findConfigFile > whith env variables > with HOME if directory exist but config file is missing
AssertionError: expected '\home\user\.config\verdaccio\config.y…' to deeply equal '/home/user/.config/verdaccio/config.y…'    

- Expected
+ Received

- /home/user/.config/verdaccio/config.yaml
+ \home\user\.config\verdaccio\config.yaml

 ❯ test/config.path.spec.ts:71:34
     69|         vi.stubEnv('XDG_CONFIG_HOME', '');
     70|         vi.stubEnv('HOME', '/home/user');
     71|         expect(findConfigFile()).toEqual('/home/user/.config/verdaccio/config.yaml');
       |                                  ^
     72|         expect(writeFileSyncMock).toHaveBeenCalledWith(
     73|           '/home/user/.config/verdaccio/config.yaml',

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/5]⎯
 FAIL  test/config.path.spec.ts > config-path > findConfigFile > whith env variables > error handling > XDG_CONFIG_HOME is not directory fallback to default
AssertionError: expected 'C:\Github\~verdaccio\verdaccio\packag…' to match 'packages/config/verdaccio/config.yaml'

- Expected
+ Received

- packages/config/verdaccio/config.yaml
+ C:\Github\~verdaccio\verdaccio\packages\config\verdaccio\config.yaml

 ❯ test/config.path.spec.ts:93:36
     91|           vi.stubEnv('XDG_CONFIG_HOME', '/home/user/fail');
     92| 
     93|           expect(findConfigFile()).toMatch('packages/config/verdaccio/config.yaml');
       |                                    ^
     94|         });
     95| 

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/5]⎯
 FAIL  test/config.path.spec.ts > config-path > findConfigFile > with no env variables > with relative location
AssertionError: expected 'C:\app\data\verdaccio\config.yaml' to match '/app/data/verdaccio/config.yaml'

- Expected
+ Received

- /app/data/verdaccio/config.yaml
+ C:\app\data\verdaccio\config.yaml

 ❯ test/config.path.spec.ts:121:34
    119|         vi.stubEnv('HOME', '');
    120|         vi.stubEnv('APPDATA', '/app/data/');
    121|         expect(findConfigFile()).toMatch('/app/data/verdaccio/config.yaml');
       |                                  ^
    122|         expect(writeFileSyncMock).toHaveBeenCalled();
    123|         expect(mkdirSyncMock).toHaveBeenCalled();

⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/5]⎯
 Test Files  1 failed | 9 passed (10)
      Tests  5 failed | 56 passed | 2 skipped | 1 todo (64)
   Start at  12:45:33
   Duration  3.06s (transform 405ms, setup 1ms, collect 3.89s, tests 216ms, environment 3ms, prepare 2.01s)
```

Ref: https://github.com/verdaccio/verdaccio/blob/master/packages/config/test/config.path.spec.ts

--- a/.changeset/swift-rabbits-vanish.md
+++ b/.changeset/swift-rabbits-vanish.md
@@ -1,0 +1,5 @@
+---
+'@verdaccio/config': patch
+---
+
+fix(config): test runs on Windows

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -125,7 +125,7 @@ describe('config-path', () => {
         vi.stubEnv('XDG_CONFIG_HOME', '');
         vi.stubEnv('HOME', '');
         vi.stubEnv('APPDATA', '/app/data/');
-        expect(findConfigFile()).toMatch(platformPath('/app/data/verdaccio/config.yaml'));
+        expect(findConfigFile()).toMatch('\\app\\data\\verdaccio\\config.yaml');
         expect(writeFileSyncMock).toHaveBeenCalled();
         expect(mkdirSyncMock).toHaveBeenCalled();
       });

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -8,6 +8,7 @@ describe('config-path', () => {
   let statSyncMock;
   let mkdirSyncMock;
   let writeFileSyncMock;
+  let platformMock;
   let accessSyncMock;
   let fakeStats = {
     isDirectory: vi.fn().mockReturnValue(true),
@@ -19,6 +20,7 @@ describe('config-path', () => {
     mkdirSyncMock = vi.spyOn(fs, 'mkdirSync');
     writeFileSyncMock = vi.spyOn(fs, 'writeFileSync');
     accessSyncMock = vi.spyOn(fs, 'accessSync');
+    platformMock = vi.spyOn(os, 'platform');
   });
 
   afterEach(() => {
@@ -95,6 +97,7 @@ describe('config-path', () => {
           expect(findConfigFile()).toMatch(platformPath('packages/config/verdaccio/config.yaml'));
         });
 
+        // Does not work on Windows
         if (os.platform() !== 'win32') {
           test('no permissions on read default config file', () => {
             vi.stubEnv('XDG_CONFIG_HOME', '/home/user');
@@ -110,13 +113,14 @@ describe('config-path', () => {
       });
     });
 
-    describe('with no env variables', () => {
+    describe('with Windows env variables', () => {
       test('with relative location', () => {
         // mock
         statSyncMock.mockReturnValue(fakeStats);
         mkdirSyncMock.mockReturnValue(true);
         writeFileSyncMock.mockReturnValue(undefined);
         accessSyncMock.mockImplementation(() => {});
+        platformMock.mockReturnValue('win32');
         // delete process.env.XDG_CONFIG_HOME;
         vi.stubEnv('XDG_CONFIG_HOME', '');
         vi.stubEnv('HOME', '');

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -8,7 +8,6 @@ describe('config-path', () => {
   let statSyncMock;
   let mkdirSyncMock;
   let writeFileSyncMock;
-  let platformMock;
   let accessSyncMock;
   let fakeStats = {
     isDirectory: vi.fn().mockReturnValue(true),
@@ -20,8 +19,6 @@ describe('config-path', () => {
     mkdirSyncMock = vi.spyOn(fs, 'mkdirSync');
     writeFileSyncMock = vi.spyOn(fs, 'writeFileSync');
     accessSyncMock = vi.spyOn(fs, 'accessSync');
-    platformMock = vi.spyOn(os, 'platform');
-    platformMock.mockReturnValue('linux');
   });
 
   afterEach(() => {
@@ -30,22 +27,27 @@ describe('config-path', () => {
     vi.unstubAllEnvs();
   });
 
+  function platformPath(path: string): string {
+    return path.replace(/\//g, os.platform() === 'win32' ? '\\' : '/');
+  }
+
   describe('findConfigFile', () => {
-    describe('using defiled location from arguments', () => {
+    describe('using file location from arguments', () => {
       test('with custom location', () => {
         // mock
         statSyncMock.mockReturnValue(fakeStats);
         mkdirSyncMock.mockReturnValue(true);
         writeFileSyncMock.mockReturnValue(undefined);
-        expect(findConfigFile('/home/user/custom/location/config.yaml')).toEqual(
-          '/home/user/custom/location/config.yaml'
+        // Note: on Windowsm, path contains drive letter
+        expect(findConfigFile('/home/user/custom/location/config.yaml')).toContain(
+          platformPath('/home/user/custom/location/config.yaml')
         );
         expect(writeFileSyncMock).not.toHaveBeenCalled();
         expect(mkdirSyncMock).not.toHaveBeenCalled();
       });
     });
 
-    describe('whith env variables', () => {
+    describe('with env variables', () => {
       test('the env XDG_CONFIG_HOME is defined and the directory exist but config file is missing', async () => {
         // mock
         statSyncMock.mockReturnValue(fakeStats);
@@ -54,9 +56,9 @@ describe('config-path', () => {
         // node env variable
         vi.stubEnv('XDG_CONFIG_HOME', '/home/user');
 
-        expect(findConfigFile()).toEqual('/home/user/verdaccio/config.yaml');
+        expect(findConfigFile()).toEqual(platformPath('/home/user/verdaccio/config.yaml'));
         expect(writeFileSyncMock).toHaveBeenCalledWith(
-          '/home/user/verdaccio/config.yaml',
+          platformPath('/home/user/verdaccio/config.yaml'),
           expect.stringContaining('packages')
         );
       });
@@ -68,13 +70,13 @@ describe('config-path', () => {
         writeFileSyncMock.mockReturnValue(undefined);
         vi.stubEnv('XDG_CONFIG_HOME', '');
         vi.stubEnv('HOME', '/home/user');
-        expect(findConfigFile()).toEqual('/home/user/.config/verdaccio/config.yaml');
+        expect(findConfigFile()).toEqual(platformPath('/home/user/.config/verdaccio/config.yaml'));
         expect(writeFileSyncMock).toHaveBeenCalledWith(
-          '/home/user/.config/verdaccio/config.yaml',
+          platformPath('/home/user/.config/verdaccio/config.yaml'),
           expect.stringContaining('packages')
         );
         expect(mkdirSyncMock).toHaveBeenCalledWith(
-          '/home/user/.config/verdaccio',
+          platformPath('/home/user/.config/verdaccio'),
           expect.anything()
         );
       });
@@ -90,19 +92,21 @@ describe('config-path', () => {
           // node env variable
           vi.stubEnv('XDG_CONFIG_HOME', '/home/user/fail');
 
-          expect(findConfigFile()).toMatch('packages/config/verdaccio/config.yaml');
+          expect(findConfigFile()).toMatch(platformPath('packages/config/verdaccio/config.yaml'));
         });
 
-        test('no permissions on read default config file', () => {
-          vi.stubEnv('XDG_CONFIG_HOME', '/home/user');
-          accessSyncMock.mockImplementation(() => {
-            throw new Error('error on write file');
+        if (os.platform() !== 'win32') {
+          test('no permissions on read default config file', () => {
+            vi.stubEnv('XDG_CONFIG_HOME', '/home/user');
+            accessSyncMock.mockImplementation(() => {
+              throw new Error('error on write file');
+            });
+
+            expect(function () {
+              findConfigFile();
+            }).toThrow(/configuration file does not have enough permissions for reading/);
           });
-
-          expect(function () {
-            findConfigFile();
-          }).toThrow(/configuration file does not have enough permissions for reading/);
-        });
+        }
       });
     });
 
@@ -113,12 +117,11 @@ describe('config-path', () => {
         mkdirSyncMock.mockReturnValue(true);
         writeFileSyncMock.mockReturnValue(undefined);
         accessSyncMock.mockImplementation(() => {});
-        platformMock.mockReturnValue('win32');
         // delete process.env.XDG_CONFIG_HOME;
         vi.stubEnv('XDG_CONFIG_HOME', '');
         vi.stubEnv('HOME', '');
         vi.stubEnv('APPDATA', '/app/data/');
-        expect(findConfigFile()).toMatch('/app/data/verdaccio/config.yaml');
+        expect(findConfigFile()).toMatch(platformPath('/app/data/verdaccio/config.yaml'));
         expect(writeFileSyncMock).toHaveBeenCalled();
         expect(mkdirSyncMock).toHaveBeenCalled();
       });

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -123,8 +123,8 @@ describe('config-path', () => {
           // delete process.env.XDG_CONFIG_HOME;
           vi.stubEnv('XDG_CONFIG_HOME', '');
           vi.stubEnv('HOME', '');
-          vi.stubEnv('APPDATA', '/app/data/');
-          expect(findConfigFile()).toMatch('\\app\\data\\verdaccio\\config.yaml');
+          vi.stubEnv('APPDATA', 'C:\\Users\\Tester\\AppData\\');
+          expect(findConfigFile()).toEqual('C:\\Users\\Tester\\AppData\\verdaccio\\config.yaml');
           expect(writeFileSyncMock).toHaveBeenCalled();
           expect(mkdirSyncMock).toHaveBeenCalled();
         });

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -8,7 +8,6 @@ describe('config-path', () => {
   let statSyncMock;
   let mkdirSyncMock;
   let writeFileSyncMock;
-  let platformMock;
   let accessSyncMock;
   let fakeStats = {
     isDirectory: vi.fn().mockReturnValue(true),
@@ -20,7 +19,6 @@ describe('config-path', () => {
     mkdirSyncMock = vi.spyOn(fs, 'mkdirSync');
     writeFileSyncMock = vi.spyOn(fs, 'writeFileSync');
     accessSyncMock = vi.spyOn(fs, 'accessSync');
-    platformMock = vi.spyOn(os, 'platform');
   });
 
   afterEach(() => {
@@ -41,7 +39,7 @@ describe('config-path', () => {
         mkdirSyncMock.mockReturnValue(true);
         writeFileSyncMock.mockReturnValue(undefined);
         // Note: on Windows, path contains drive letter
-        expect(findConfigFile('/home/user/custom/location/config.yaml')).toContain(
+        expect(findConfigFile('/home/user/custom/location/config.yaml')).toMatch(
           platformPath('/home/user/custom/location/config.yaml')
         );
         expect(writeFileSyncMock).not.toHaveBeenCalled();
@@ -113,22 +111,24 @@ describe('config-path', () => {
       });
     });
 
-    describe('with Windows env variables', () => {
-      test('with relative location', () => {
-        // mock
-        statSyncMock.mockReturnValue(fakeStats);
-        mkdirSyncMock.mockReturnValue(true);
-        writeFileSyncMock.mockReturnValue(undefined);
-        accessSyncMock.mockImplementation(() => {});
-        platformMock.mockReturnValue('win32');
-        // delete process.env.XDG_CONFIG_HOME;
-        vi.stubEnv('XDG_CONFIG_HOME', '');
-        vi.stubEnv('HOME', '');
-        vi.stubEnv('APPDATA', '/app/data/');
-        expect(findConfigFile()).toMatch(platformPath('/app/data/verdaccio/config.yaml'));
-        expect(writeFileSyncMock).toHaveBeenCalled();
-        expect(mkdirSyncMock).toHaveBeenCalled();
+    // Note: Trying to mock Windows platform leads to different results (incorrect slashes)
+    if (os.platform() === 'win32') {
+      describe('with Windows env variables', () => {
+        test('with relative location', () => {
+          // mock
+          statSyncMock.mockReturnValue(fakeStats);
+          mkdirSyncMock.mockReturnValue(true);
+          writeFileSyncMock.mockReturnValue(undefined);
+          accessSyncMock.mockImplementation(() => {});
+          // delete process.env.XDG_CONFIG_HOME;
+          vi.stubEnv('XDG_CONFIG_HOME', '');
+          vi.stubEnv('HOME', '');
+          vi.stubEnv('APPDATA', '/app/data/');
+          expect(findConfigFile()).toMatch('\\app\\data\\verdaccio\\config.yaml');
+          expect(writeFileSyncMock).toHaveBeenCalled();
+          expect(mkdirSyncMock).toHaveBeenCalled();
+        });
       });
-    });
+    }
   });
 });

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -125,7 +125,7 @@ describe('config-path', () => {
         vi.stubEnv('XDG_CONFIG_HOME', '');
         vi.stubEnv('HOME', '');
         vi.stubEnv('APPDATA', '/app/data/');
-        expect(findConfigFile()).toMatch('\\app\\data\\verdaccio\\config.yaml');
+        expect(findConfigFile()).toMatch(platformPath('/app/data/verdaccio/config.yaml'));
         expect(writeFileSyncMock).toHaveBeenCalled();
         expect(mkdirSyncMock).toHaveBeenCalled();
       });

--- a/packages/config/test/config.path.spec.ts
+++ b/packages/config/test/config.path.spec.ts
@@ -38,7 +38,7 @@ describe('config-path', () => {
         statSyncMock.mockReturnValue(fakeStats);
         mkdirSyncMock.mockReturnValue(true);
         writeFileSyncMock.mockReturnValue(undefined);
-        // Note: on Windowsm, path contains drive letter
+        // Note: on Windows, path contains drive letter
         expect(findConfigFile('/home/user/custom/location/config.yaml')).toContain(
           platformPath('/home/user/custom/location/config.yaml')
         );

